src/main/java/org/apache/commons/collections4/ListUtils.java

```diff
@@ -173,6 +174,36 @@ public static <E> List<E> fixedSizeList(final List<E> list) {
         return FixedSizeList.fixedSizeList(list);
     }
 
+    /**
+     * Finds and returns a list of duplicate elements in the given list.
+     * <p>
+     * This method uses two sets: one for tracking seen elements and one for
+     * collecting duplicates. It iterates through the list once and collects
+     * duplicates in a result list.
+     * </p>
+     *
+     * @param <E> the type of elements in the list
+     * @param list the list to check for duplicates, must not be null
+     * @return a list of duplicate elements, or an empty list if no duplicates are found
+     * @throws NullPointerException if the list is null
+     * @since 4.5.0-M3
+     */
+    public static <E> List<E> findDuplicates(final List<E> list) {
+        Objects.requireNonNull(list, "The input list must not be null.");
```

Comment:
@hemanth0525 

If you replace the body of this (git master) method with:
```java
return IterableUtils.duplicateList(list);
```
You'll find that all tests pass.
So I don't think we need this method in this class.

Reply:
u mean from CollectionUtils ?

Reply:
if yes, I have done it. So yeah utilizing the other function to help list functionalities.